File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
      Map<Integer,Integer> mp = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            int req = target - nums[i];
            if(mp.containsKey(req)){
                int []ans = {i, mp.get(req)};
                return ans;
            }
            mp.put(nums[i], i);
        }
        return new int [0];
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp;
        int n = nums.size();
        for(int i=0; i<n; i++){
            mp[nums[i]] = i;
        }
        
        for(int i=0; i<n; i++){
            int req = target - nums[i];
            if(mp.find(req) != mp.end() && mp[req] != i) return {mp[req], i};
        }
        
        
        return {-1, -1};
    }
};